library/src/main/java/com/novoda/downloadmanager/FileDownloader.java

```diff
@@ -23,8 +23,41 @@
 
         void onBytesRead(byte[] buffer, int bytesRead);
 
-        void onError(String cause);
+        void onError(FileDownloadError cause);
 
         void onDownloadFinished();
     }
+
+    /**
+     * Represents the information of a {@link LiteFileDownloadError} that is accessible to clients.
+     */
+    interface FileDownloadError {
+
+        /**
+         * Use to create instances of {@link LiteFileDownloadError}.
+         *
+         * @param rawRequest the raw request from which the error occurs.
+         * @param message    any message associated with the error.
+         * @param errorCode  the code associated with the error.
+         * @return an instance of {@link FileDownloadError}.
+         */
+        static FileDownloader.FileDownloadError createFrom(String rawRequest, String message, int errorCode) {
+            return new LiteFileDownloadError(rawRequest, message, errorCode);
+        }
+
+        /**
+         * @return the raw request performed when the error occurred.
+         */
+        String rawRequest();
```

Comment:
This could be something like `requestUrl`?
Raw request sounds like the full okhttp request (headers and ecc...)

Reply:
The problem is that the `FileDownloader` may not necessarily call the network, I wasn't really sure about `rawRequest`, in my mind it should probably be `fileDownloadPath` to keep it agnostic.

Reply:
Wearing the "I don't know this library user" hat here:
TBH `fileDownloadPath` sounds like the path where the file is gonna be downloaded.
And looking at the `FileDownloader.startDownloading(..)` signature, the first param is named url...

Reply:
That's a fair point. I'll rename both so that they are `requestUrl` then 😄

Reply:
cb5db7cb07ca44459a0c4306690d5a6a3bc06e26